test_gpt_researcher.py

```diff
@@ -0,0 +1,50 @@
+import os
+import asyncio
+import pytest
+from gpt_researcher import GPTResearcher
+
+# Define the report types to test
+report_types = [
+    "research_report",
+    "custom_report",
+    "subtopic_report",
+    "summary_report",
+    "detailed_report",
+    "quick_report"
+]
+
+# Define a common query and sources for testing
+query = "What are the latest advancements in AI?"
+sources = ["https://en.wikipedia.org/wiki/Artificial_intelligence", "https://www.ibm.com/watson/ai"]
+
+# Define the output directory
+output_dir = "./outputs"
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("report_type", report_types)
+async def test_gpt_researcher(report_type):
+    # Ensure the output directory exists
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+        
+    # Create an instance of GPTResearcher
+    researcher = GPTResearcher(query=query, report_type=report_type, source_urls=sources)
+    
+    # Conduct research and write the report
+    await researcher.conduct_research()
+    report = await researcher.write_report()
+    
+    # Define the expected output filenames
+    pdf_filename = os.path.join(output_dir, f"{report_type}.pdf")
+    docx_filename = os.path.join(output_dir, f"{report_type}.docx")
+    
+    # Check if the PDF and DOCX files are created
+    # assert os.path.exists(pdf_filename), f"PDF file not found for report type: {report_type}"
```

Comment:
@ElishaKay why is this commented? Is this test final?

Reply:
For those 6 use cases in the commit message above, correct, the tests are final and pass.

I need to make a change in agent.py for those tests to pass (`initialize report = ""` at top of the function. If that doesn't break anything feel free to merge. Else we can continue writing tests for more use cases here.

Food.for thought:

 - What are the potential parameters and values that can be passed to the GPTResearcher Class?
 - For end to end tests (fronted to backend and back), we'd be better positioned leveraging a React Testing Library

Reply:
Just got me thinking that GPT Researcher should be able to expose the writing functions